src/main/java/io/dropwizard/revolver/RevolverBundle.java

```diff
@@ -271,6 +271,7 @@ private void initializeRevolver(final T configuration, final Environment environ
     }
 
     public static void loadServiceConfiguration(RevolverConfig revolverConfig) {
+        serviceToPathMap = new MultivaluedHashMap<>();
```

Comment:
At evert app config refresh, the serviceToPathMap is getting filled with same configs again and again. At one point, this map will bloat. Moreover for any config update, this map will contain both old and new config which is quite dangerous

Reply:
Erasing this map at loadtime will lead to lookup failures till the map is completely loaded. We cannot erase this map.

Reply:
We will have to do something like this:
`    private static Map<String, RevolverHttpApiConfig> generateApiConfigMap(final RevolverHttpServiceConfig serviceConfiguration) {
        val tokenMatch = Pattern.compile("\\{(([^/])+\\})");
        List<RevolverHttpApiConfig> apis = new ArrayList<>(serviceConfiguration.getApis());
        apis.sort((o1, o2) -> {
            String o1Expr = generatePathExpression(o1.getPath());
            String o2Expr = generatePathExpression(o2.getPath());
            return tokenMatch.matcher(o2Expr).groupCount() - tokenMatch.matcher(o1Expr).groupCount();
        });
        apis.sort(Comparator.comparing(RevolverHttpApiConfig::getPath));
        apis.forEach(apiConfig -> {
            final ApiPathMap apiPathMap = ApiPathMap.builder()
                    .api(apiConfig)
                    .path(generatePathExpression(apiConfig.getPath())).build();
            //Update
            int elementIndex = serviceToPathMap.getOrDefault(serviceConfiguration.getService(), Collections.emptyList())
                    .indexOf(apiPathMap);
            if(elementIndex == -1) {
                serviceToPathMap.add(serviceConfiguration.getService(), apiPathMap);
            } else {
                serviceToPathMap.get(serviceConfiguration.getService()).set(elementIndex, apiPathMap);
            }
        });
        final ImmutableMap.Builder<String, RevolverHttpApiConfig> configMapBuilder = ImmutableMap.builder();
        apis.forEach(apiConfig -> configMapBuilder.put(apiConfig.getApi(), apiConfig));
        return configMapBuilder.build();
    }`

Reply:
@phaneesh Fixed as you said